math-lib/math/private/number-theory/quadratic.rkt

```diff
@@ -4,32 +4,61 @@
          (only-in racket/math conjugate)
          "types.rkt")
 
-(provide quadratic-complex-solutions
-         quadratic-solutions
-         quadratic-integer-solutions
-         quadratic-natural-solutions)
+(provide complex-quadratic-solutions  ; complex coefficients
+         quadratic-solutions          ; real coefficients
+         quadratic-integer-solutions  ; coefficients
+         quadratic-natural-solutions) ;coefficients
 
-(: quadratic-complex-solutions  : Complex Complex Complex -> (Listof Complex))
-(define (quadratic-complex-solutions a b c)
+
+(: complex-quadratic-solutions  : Complex Complex Complex -> (Listof Complex))
+(define (complex-quadratic-solutions a b c)
+  ; Return list of solutions to a a x^2 + b x + c = 0
+  ; where a,b and c are complex numbers.
+  (let* ([d         (- (* b b) (* 4 a c))]
+         [sqrt-d    (sqrt d)]
+         [-b-sqrt-d (- (- b) sqrt-d)]
+         [2a        (* 2 a)])
+    (cond
+      [(= d 0) (list (/ b (- 2a)))]
+      [(= c 0) (list (/ (- (- b) sqrt-d) 2a)
+                     (/ (+ (- b) sqrt-d) 2a))]
+      ; use the standard formula, unless -b and sqrt are almost equal
+      [#t ; (> (magnitude -b-sqrt-d) 0.001)
+       (displayln d)
```

Comment:
Was this meant to stay in the final code?

Reply:
No. Thanks for noticing.